### PR TITLE
Add pollChatPosts() to the API.

### PR DESCRIPTION
**Pull request by Red-exe-Engineer**
I found out how to create the pull request. xD
(Hope this is everything!)

### CODE REVIEW

File: mcpi_addons/minecraft.py [hunk 1]
```diff
@@ -186,6 +186,18 @@ def __init__(self, connection):
     def clearAll(self):
         """Clear all old events"""
         self.conn.send(b"events.clear")
+        
+    def pollChatPosts(self):
+        s = self.conn.sendReceive(b"events.pollChatPosts").split("\0")
```
Comment (Bigjango13) [inline mcpi_addons/minecraft.py]: This should be `events.chat.posts`

File: src/api.cpp [hunk 2]
```diff
@@ -2,20 +2,26 @@
 #include <fstream>
 #include <map>
 #include <string>
+#include <vector>
```
Comment (Bigjango13) [inline src/api.cpp]: `<vector>` is already included by `<symbols/minecraft.h>`

File: src/api.cpp [hunk 3]
```diff
@@ -2,20 +2,26 @@
...
+#include <symbols/minecraft.h>
 #include <mods/misc/misc.h>
+#include <mods/chat/chat.h>
 
 #include "api.h"
 #include "base64.h"
 #include "extra.h"
 #include "helpers.h"
 
+std::vector<std::string> chat_log = {};
```
Comment (Bigjango13) [inline src/api.cpp:17]: Could this be moved closer to where it's actually used?

File: src/extra.h [hunk 4]
```diff
@@ -18,7 +18,7 @@ static Player_getArmor_t Player_getArmor = (Player_getArmor_t) 0x8fda4;
 typedef void (*offsetCords_t)(uchar *offsetData, int *x, int *y, int *z);
 static offsetCords_t offsetCords = (offsetCords_t) 0x27c98;
 
-static uchar **Item_items = (uchar **) 0x17b250;
+static uchar **Item_items = (uchar **) 0x17b250; // Comment this out if you're using MCPI-Reborn Extended!
```
Comment (Bigjango13) [inline src/extra.h:21]: Could you move it to not be an inline comment?

File: src/api.cpp [hunk 5]
```diff
@@ -392,6 +398,34 @@ std::string handle_entity(std::string command, std::string args, uchar *command_
...
+}
+
+HOOK(chat_send_message, void, (unsigned char *server_side_network_handler, char *username, char *message)) {
+    std::string log = std::string(username) + '\0' + std::string(message);
+    chat_log.push_back(log);
+    ensure_chat_send_message();
+    (*real_chat_send_message)(server_side_network_handler, username, message);
+
+    if (chat_log.size() > 64) {
+        chat_log.erase(chat_log.begin(), chat_log.begin() + (chat_log.size() - 3));
```
Comment (Bigjango13) [inline src/api.cpp]: I'm not sure what this does, a comment would be very helpful

**Review by Bigjango13 — CHANGES_REQUESTED** [review]
Looks good, some minor style/implementation things.

File: mcpi_addons/minecraft.py (hunk 1 above)
Comment (Red-exe-Engineer) [reply mcpi_addons/minecraft.py]: This causes the API to hang, might have something to do with the use of two dots being used?

File: src/api.cpp (hunk 3 above)
Comment (Red-exe-Engineer) [reply src/api.cpp:17]: I put it there because other stuff was being defined in that area but sure, I'll do that.

Comment (Red-exe-Engineer) [reply src/api.cpp:17]: Nevermind! I forgot that I must define stuff before I define the function, and putting it between the `handle_pollChatPosts` and previous functions looks weird.

File: src/extra.h (hunk 4 above)
Comment (Red-exe-Engineer) [reply src/extra.h:21]: I put it after the line so if it errors it will show the comment too.

File: src/api.cpp [hunk 6]
```diff
@@ -392,6 +398,48 @@ std::string handle_entity(std::string command, std::string args, uchar *command_
...
+    } catch (const std::invalid_argument& e) {
+        return "\n";
+    }
+}
+
+std::string handle_getVersion(std::string command, std::string args, uchar *command_server) {
+    return std::string(*minecraft_pi_version) + "\n";
+}
+
+HOOK(chat_send_message, void, (unsigned char *server_side_network_handler, char *username, char *message)) {
```
Comment (TheBrokenRail) [inline src/api.cpp:443]: This works on servers/the hosting player, but this function isn't called on clients.

Comment (Red-exe-Engineer) [reply src/api.cpp:443]: I don't know of another easy-to-hook function that doesn't require some form of parsing. :/

Comment (Bigjango13) [reply src/api.cpp:443]: `Gui::renderChatMessages` displays the chat, maybe that can be used to get the messages?

File: mcpi_addons/minecraft.py [hunk 7]
```diff
@@ -232,14 +232,14 @@ def give(self, id=-2, auxiliary=-2, count=-2):
 class CmdReborn:
     """Reborn"""
 
-    def __init__(self, connection):
```
Comment (Red-exe-Engineer) [inline mcpi_addons/minecraft.py:219]: Oops!

File: mcpi_addons/minecraft.py (hunk 1 above)
Comment (Red-exe-Engineer) [reply mcpi_addons/minecraft.py]: I figured out how to do this!

File: src/api.cpp (hunk 6 above)
Comment (Bigjango13) [reply src/api.cpp:443]: `Gui::addMessage` should be used, but currently can't (without disabling Reborn stuff) until a bug with `extract_from_bl_instruction` is fixed (or a workaround is found).

**Pull request merged**